Testscripts/Windows/LIS-Mount-CD.ps1

```diff
@@ -0,0 +1,146 @@
+    # Check arguments
+    if (-not $VMName) {
+       LogErr "Missing vmName argument"
+        return $False
+    }
+
+    if (-not $HvServer) {
+       LogErr  "Missing hvServer argument"
+        return $False
+    }
+    #
+    # Extract the testParams
+    #
+    $params = $TestParams.Split(';')
+    foreach ($p in $params) {
+        if ($p.Trim().Length -eq 0) {
+            continue
+        }
+        $tokens = $p.Trim().Split('=')
+
+        if ($tokens.Length -ne 2) {
+            # Just ignore it
+            continue
+        }
+
+        $lValue = $tokens[0].Trim()
+        $rValue = $tokens[1].Trim()
+
+        if ($lValue -eq "HotAdd") {
+            $hotAdd = $rValue
+        }
+    }
+
+    $error.Clear()
+
+    $vmGen = Get-VMGeneration  -vmName  $VMName -hvServer $HvServer
+    if ( $hotAdd -eq "True" -and $vmGen -eq 1) {
+       LogMsg " Generation 1 VM does not support hot add DVD, please skip this case in the test script"
+        return $True
+    }
+
+    #
+    # There should be only one DVD unit by default
+    #
+    $dvd = Get-VMDvdDrive $VMName -ComputerName $HvServer
+    if ( $dvd ) {
+        try {
+            Remove-VMDvdDrive $dvd -Confirm:$False
+        }
+        catch {
+            LogErr "Cannot remove DVD drive from ${vmName}"
+            $error[0].Exception
+            return $False
+        }
+    }
+
+    #
+    # Get Hyper-V VHD path
+    #
+    $obj = Get-WmiObject -ComputerName $HvServer -Namespace "root\virtualization\v2" -Class "MsVM_VirtualSystemManagementServiceSettingData"
+    $defaultVhdPath = $obj.DefaultVirtualHardDiskPath
+    if (-not $defaultVhdPath) {
+        LogErr "Unable to determine VhdDefaultPath on Hyper-V server ${hvServer}"
+        $error[0].Exception
+        return $False
+    }
+    if (-not $defaultVhdPath.EndsWith("\")) {
+        $defaultVhdPath += "\"
+    }
+    $isoPath = $defaultVhdPath + "${vmName}_CDtest.iso"
+
+    $WebClient = New-Object System.Net.WebClient
+    $WebClient.DownloadFile("$url", "$isoPath")
+
+    try {
+        Get-RemoteFileInfo -filename $isoPath  -server $HvServer
+    }
+    catch {
+        LogErr "The .iso file $isoPath could not be found!"
+        return $False
+    }
+
+    #
+    # Insert the .iso file into the VMs DVD drive
+    #
+    if ($vmGen -eq 1) {
+        Add-VMDvdDrive -VMName $VMName -Path $isoPath -ControllerNumber 1 -ControllerLocation 1 -ComputerName $HvServer -Confirm:$False
+    }
+    else {
+        Add-VMDvdDrive -VMName $VMName -Path $isoPath -ControllerNumber 0 -ControllerLocation 1 -ComputerName $HvServer -Confirm:$False
+    }
+
+    if ($? -ne "True") {
+        LogErr "Unable to mount the ISO file!"
+        $error[0].Exception
+        return $False
+    }
+    else {
+        $retVal = $True
+    }
+
+    return $retVal
+}
+Main -VMName $AllVMData.RoleName -HvServer $xmlConfig.config.Hyperv.Host.ServerName `
+    -Ipv4 $AllVMData.PublicIP -VMPort $AllVMData.SSHPort `
```

Comment:
You don't seem to need ipv4, VmPort, VmUsername, VmPassword or RootDir params in this powershell script. In this case, scrap them, and use just VMName, HvServer & TestParams params